src/lib/float32_truncate.js

```diff
@@ -0,0 +1,21 @@
+/**
+* Copyright 2012-2016, Plotly, Inc.
+* All rights reserved.
+*
+* This source code is licensed under the MIT license found in the
+* LICENSE file in the root directory of this source tree.
+*/
+
+'use strict';
+
+/**
+ * Truncate a Float32Array to some length. A wrapper to support environments
+ * (e.g. node-webkit) that do not implement Float32Array.prototype.slice
+ */
+module.exports = function truncate(float32ArrayIn, len) {
+    // for some reason, ES2015 Float32Array.prototype.slice takes 2x as long...
+    // therefore we aren't checking for its existence
```

Comment:
All too often, 'recently' added language elements (as in: the last few years) are slower even if intuitively they should be faster. Back in 2014, untyped arrays were faster than typed arrays even for plain calculations; same with `Map`.